File: easy/find_lucky_test.go
package easy

import (
	"testing"
)

func TestFindLucky(t *testing.T) {
	cases := []struct {
		nums     []int
		expected int
	}{
		{[]int{2, 2, 3, 4}, 2},
		{[]int{1, 2, 2, 3, 3, 3}, 3},
		{[]int{2, 2, 2, 3, 3}, -1},
		{[]int{5, 5, 5, 5, 5}, 5},
		{[]int{7, 7, 7, 7, 7, 7}, -1},
	}
	for _, c := range cases {
		t.Run("", func(t *testing.T) {
			result := FindLucky(c.nums)
			if result != c.expected {
				t.Errorf("FindLucky(%v) == %v, expected %v", c.nums, result, c.expected)
			}

		})

	}
}

File: src/test/java/easy/FindLuckyTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class FindLuckyTest {
  FindLucky instance = new FindLucky();

  @ParameterizedTest
  @MethodSource("dataFindLucky")
  public void testFindLucky(int[] nums, int expected) {

    int result = instance.solve(nums);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataFindLucky() {
    return Stream.of(
        Arguments.of(new int[] {2, 2, 3, 4}, 2),
        Arguments.of(new int[] {1, 2, 2, 3, 3, 3}, 3),
        Arguments.of(new int[] {2, 2, 2, 3, 3}, -1));
  }
}
